lib/modules/datasource/docker/common.ts

```diff
@@ -144,23 +146,20 @@ export async function getAuthHeaders(
     ) {
       scope = authenticateHeader.params.scope;
     }
+    authUrl.searchParams.append('scope', scope);
 
-    let service = authenticateHeader.params.service;
-    if (is.string(service)) {
-      service = `service=${service}&`;
-    } else {
-      service = ``;
+    if (is.string(authenticateHeader.params.service)) {
+      authUrl.searchParams.append('service', authenticateHeader.params.service);
     }
 
-    const authUrl = `${authenticateHeader.params.realm}?${service}scope=${scope}`;
     logger.trace(
-      { registryHost, dockerRepository, authUrl },
+      { registryHost, dockerRepository, url: authUrl.href },
```

Comment:
```suggestion
      { registryHost, dockerRepository, authUrl: authUrl.href },
```
don't change the name